[qt5-webengine] assert vc_lib_atlmfc_path AssertionError

**Host Environment**
- OS: Window 10 (x64-windows)
- Compiler: The CXX compiler identification is MSVC 19.25.28614.0. Visual Studio 2019 Developer Command Prompt v16.5.4. Windows SDK version 10.0.18362.0 to target Windows 10.0.18363.

**To Reproduce**

`vcpkg install qt5-base:x64-windows qt5-webengine:x64-windows`

**Failure logs**

package-build-x64-windows-rel-out.log :

```console
cd src\ && ( if not exist Makefile D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -o Makefile D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\src.pro "CONFIG-=debug" "CONFIG+=release" "CONFIG-=static" "CONFIG*=shared" -qtconf D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/qt.conf ) && D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile
	cd buildtools\ && ( if not exist Makefile D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -o Makefile D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\buildtools\buildtools.pro "CONFIG-=debug" "CONFIG+=release" "CONFIG-=static" "CONFIG*=shared" -qtconf D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/qt.conf ) && D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile
	( if not exist Makefile.ninja D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -o Makefile.ninja D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\buildtools\ninja.pro "CONFIG-=debug" "CONFIG+=release" "CONFIG-=static" "CONFIG*=shared" -qtconf D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/qt.conf ) && D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.ninja
	cd webengine\ui\ && ( if not exist Makefile D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -o Makefile D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\webengine\ui\ui.pro "CONFIG-=debug" "CONFIG+=release" "CONFIG-=static" "CONFIG*=shared" -qtconf D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/qt.conf ) && D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile
	D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.Release
	D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -install qinstall D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\webengine\ui\qmldir ..\..\..\qml\QtWebEngine\Controls1Delegates\qmldir
	cd webengine\ui2\ && ( if not exist Makefile D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -o Makefile D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\webengine\ui2\ui2.pro "CONFIG-=debug" "CONFIG+=release" "CONFIG-=static" "CONFIG*=shared" -qtconf D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/qt.conf ) && D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile
	D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.Release
	D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -install qinstall D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\webengine\ui2\qmldir ..\..\..\qml\QtWebEngine\Controls2Delegates\qmldir
build.cc
build_log.cc
clean.cc
clparser.cc
debug_flags.cc
depfile_parser.cc
deps_log.cc
disk_interface.cc
edit_distance.cc
eval_env.cc
graph.cc
graphviz.cc
lexer.cc
line_printer.cc
manifest_parser.cc
metrics.cc
state.cc
string_piece_util.cc
util.cc
version.cc
subprocess-win32.cc
includes_normalize-win32.cc
msvc_helper-win32.cc
msvc_helper_main-win32.cc
minidump-win32.cc
getopt.c
ninja.cc
Generating code
Finished generating code
[1/29] CXX build\debug_flags.obj
[2/29] CXX build\version.obj
[3/29] CXX build\depfile_parser.obj
[4/29] CXX build\edit_distance.obj
[5/29] CXX build\clparser.obj
[6/29] CXX build\deps_log.obj
[7/29] CXX build\manifest_parser.obj
[8/29] CXX build\build_log.obj
[9/29] CXX build\clean.obj
[10/29] CXX build\graphviz.obj
[11/29] CXX build\eval_env.obj
[12/29] CXX build\graph.obj
[13/29] CXX build\lexer.obj
[14/29] CXX build\string_piece_util.obj
[15/29] CXX build\subprocess-win32.obj
[16/29] CXX build\getopt.obj
[17/29] CXX build\state.obj
[18/29] CXX build\build.obj
[19/29] CXX build\line_printer.obj
[20/29] CXX build\metrics.obj
[21/29] CXX build\disk_interface.obj
[22/29] CXX build\util.obj
[23/29] CXX build\minidump-win32.obj
[24/29] CXX build\msvc_helper_main-win32.obj
[25/29] CXX build\msvc_helper-win32.obj
[26/29] CXX build\ninja.obj
[27/29] CXX build\includes_normalize-win32.obj
[28/29] LIB build\ninja.lib
[29/29] LINK ninja.exe
Generating code
Finished generating code
bootstrapping ninja...
warning: A compatible version of re2c (>= 0.11.3) was not found; changes to src/*.in.cc will not affect your build.
wrote build.ninja.
bootstrap complete.  rebuilding...
	D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.ninja.Release
	( if not exist Makefile.gn D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -o Makefile.gn D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\buildtools\gn.pro "CONFIG-=debug" "CONFIG+=release" "CONFIG-=static" "CONFIG*=shared" -qtconf D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/qt.conf ) && D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.gn
[1/172] CXX base/files/file_path_constants.obj
[2/172] CXX base/callback_internal.obj
[3/172] CXX base/files/scoped_file.obj
[4/172] CXX base/files/file.obj
[5/172] CXX base/files/scoped_temp_dir.obj
[6/172] CXX base/json/json_parser.obj
[7/172] CXX base/environment.obj
[8/172] CXX base/files/file_path.obj
[9/172] CXX base/files/file_enumerator.obj
[10/172] CXX base/command_line.obj
[11/172] CXX base/memory/ref_counted.obj
[12/172] CXX base/memory/weak_ptr.obj
[13/172] CXX base/json/json_writer.obj
[14/172] CXX base/md5.obj
[15/172] CXX base/json/string_escape.obj
[16/172] CXX base/strings/string_util_constants.obj
[17/172] CXX base/json/json_reader.obj
[18/172] CXX base/files/file_util.obj
[19/172] CXX base/third_party/icu/icu_utf.obj
[20/172] CXX base/strings/stringprintf.obj
[21/172] CXX base/strings/string_piece.obj
[22/172] CXX base/sha1.obj
[23/172] CXX base/timer/elapsed_timer.obj
[24/172] CXX base/strings/string_split.obj
[25/172] CXX base/logging.obj
[26/172] CXX base/strings/string_util.obj
[27/172] CXX base/strings/utf_string_conversion_utils.obj
[28/172] CXX base/value_iterators.obj
[29/172] CXX base/strings/string_number_conversions.obj
[30/172] CXX base/strings/utf_string_conversions.obj
[31/172] CXX base/values.obj
[32/172] CXX base/strings/sys_string_conversions_win.obj
[33/172] CXX base/files/file_enumerator_win.obj
[34/172] CXX base/win/scoped_process_information.obj
[35/172] CXX base/files/file_win.obj
[36/172] CXX base/win/scoped_handle.obj
[37/172] CXX base/win/registry.obj
[38/172] CXX base/files/file_util_win.obj
[39/172] CXX tools/gn/action_values.obj
[40/172] CXX tools/gn/action_target_generator.obj
[41/172] CXX tools/gn/args.obj
[42/172] CXX tools/gn/analyzer.obj
[43/172] CXX tools/gn/builder_record.obj
[44/172] CXX tools/gn/bundle_file_rule.obj
[45/172] CXX tools/gn/binary_target_generator.obj
[46/172] CXX tools/gn/bundle_data.obj
[47/172] CXX tools/gn/build_settings.obj
[48/172] LIB base.lib
[49/172] CXX tools/gn/c_include_iterator.obj
[50/172] CXX tools/gn/bundle_data_target_generator.obj
[51/172] CXX tools/gn/builder.obj
[52/172] CXX tools/gn/command_args.obj
[53/172] CXX tools/gn/command_check.obj
[54/172] CXX tools/gn/command_clean.obj
[55/172] CXX tools/gn/command_format.obj
[56/172] CXX tools/gn/compile_commands_writer.obj
[57/172] CXX tools/gn/command_analyze.obj
[58/172] CXX tools/gn/command_ls.obj
[59/172] CXX tools/gn/command_desc.obj
[60/172] CXX tools/gn/command_gen.obj
[61/172] CXX tools/gn/config_values.obj
[62/172] CXX tools/gn/command_refs.obj
[63/172] CXX tools/gn/exec_process.obj
[64/172] CXX tools/gn/config_values_extractors.obj
[65/172] CXX tools/gn/command_path.obj
[66/172] CXX tools/gn/command_help.obj
[67/172] CXX tools/gn/copy_target_generator.obj
[68/172] CXX tools/gn/commands.obj
[69/172] CXX tools/gn/escape.obj
[70/172] CXX tools/gn/err.obj
[71/172] CXX tools/gn/eclipse_writer.obj
[72/172] CXX tools/gn/create_bundle_target_generator.obj
[73/172] CXX tools/gn/config.obj
[74/172] CXX tools/gn/config_values_generator.obj
[75/172] CXX tools/gn/function_template.obj
[76/172] CXX tools/gn/function_foreach.obj
[77/172] CXX tools/gn/function_forward_variables_from.obj
[78/172] CXX tools/gn/deps_iterator.obj
[79/172] CXX tools/gn/desc_builder.obj
[80/172] CXX tools/gn/function_process_file_template.obj
[81/172] CXX tools/gn/filesystem_utils.obj
[82/172] CXX tools/gn/function_exec_script.obj
[83/172] CXX tools/gn/function_get_target_outputs.obj
[84/172] CXX tools/gn/function_set_defaults.obj
[85/172] CXX tools/gn/function_get_label_info.obj
[86/172] CXX tools/gn/function_rebase_path.obj
[87/172] CXX tools/gn/function_get_path_info.obj
[88/172] CXX tools/gn/function_read_file.obj
[89/172] CXX tools/gn/function_set_default_toolchain.obj
[90/172] CXX tools/gn/functions_target.obj
[91/172] CXX tools/gn/functions.obj
[92/172] CXX tools/gn/group_target_generator.obj
[93/172] CXX tools/gn/item.obj
[94/172] CXX tools/gn/label.obj
[95/172] CXX tools/gn/function_toolchain.obj
[96/172] CXX tools/gn/json_project_writer.obj
[97/172] CXX tools/gn/input_file.obj
[98/172] CXX tools/gn/input_file_manager.obj
[99/172] CXX tools/gn/lib_file.obj
[100/172] CXX tools/gn/inherited_libraries.obj
[101/172] CXX tools/gn/import_manager.obj
[102/172] CXX tools/gn/location.obj
[103/172] CXX tools/gn/header_checker.obj
[104/172] CXX tools/gn/label_pattern.obj
[105/172] CXX tools/gn/function_write_file.obj
[106/172] CXX tools/gn/operators.obj
[107/172] CXX tools/gn/input_conversion.obj
[108/172] CXX tools/gn/ninja_bundle_data_target_writer.obj
[109/172] CXX tools/gn/ninja_action_target_writer.obj
[110/172] CXX tools/gn/loader.obj
[111/172] CXX tools/gn/ninja_target_command_util.obj
[112/172] CXX tools/gn/ninja_group_target_writer.obj
[113/172] CXX tools/gn/ninja_binary_target_writer.obj
[114/172] CXX tools/gn/ninja_copy_target_writer.obj
[115/172] CXX tools/gn/pool.obj
[116/172] CXX tools/gn/ninja_create_bundle_target_writer.obj
[117/172] CXX tools/gn/ninja_utils.obj
[118/172] CXX tools/gn/output_conversion.obj
[119/172] CXX tools/gn/ninja_toolchain_writer.obj
[120/172] CXX tools/gn/ninja_writer.obj
[121/172] CXX tools/gn/pattern.obj
[122/172] CXX tools/gn/parse_node_value_adapter.obj
[123/172] CXX tools/gn/ninja_target_writer.obj
[124/172] CXX tools/gn/output_file.obj
[125/172] CXX tools/gn/parser.obj
[126/172] CXX tools/gn/ninja_build_writer.obj
[127/172] CXX tools/gn/parse_tree.obj
[128/172] CXX tools/gn/path_output.obj
[129/172] CXX tools/gn/scope.obj
[130/172] CXX tools/gn/qt_creator_writer.obj
[131/172] CXX tools/gn/scheduler.obj
[132/172] CXX tools/gn/scope_per_file_provider.obj
[133/172] CXX tools/gn/qmake_link_writer.obj
[134/172] CXX tools/gn/standard_out.obj
[135/172] CXX tools/gn/substitution_pattern.obj
[136/172] CXX tools/gn/source_dir.obj
[137/172] CXX tools/gn/runtime_deps.obj
[138/172] CXX tools/gn/switches.obj
[139/172] CXX tools/gn/substitution_list.obj
[140/172] CXX tools/gn/settings.obj
[141/172] CXX tools/gn/substitution_type.obj
[142/172] CXX tools/gn/substitution_writer.obj
[143/172] CXX tools/gn/token.obj
[144/172] CXX tools/gn/string_utils.obj
[145/172] CXX tools/gn/setup.obj
[146/172] CXX tools/gn/source_file.obj
[147/172] CXX tools/gn/source_file_type.obj
[148/172] CXX tools/gn/xml_element_writer.obj
[149/172] CXX tools/gn/variables.obj
[150/172] CXX tools/gn/tool.obj
[151/172] CXX tools/gn/template.obj
[152/172] CXX tools/gn/target.obj
[153/172] CXX tools/gn/visual_studio_utils.obj
[154/172] CXX tools/gn/target_generator.obj
[155/172] CXX tools/gn/value_extractors.obj
[156/172] CXX tools/gn/tokenizer.obj
[157/172] CXX tools/gn/toolchain.obj
[158/172] CXX tools/gn/value.obj
[159/172] CXX tools/gn/trace.obj
[160/172] CXX tools/gn/xcode_object.obj
[161/172] CXX tools/gn/visual_studio_writer.obj
[162/172] CXX util/worker_pool.obj
[163/172] CXX util/sys_info.obj
[164/172] CXX util/ticks.obj
[165/172] CXX util/exe_path.obj
[166/172] CXX tools/gn/visibility.obj
[167/172] CXX tools/gn/xcode_writer.obj
[168/172] CXX util/msg_loop.obj
[169/172] CXX util/semaphore.obj
[170/172] CXX tools/gn/gn_main.obj
[171/172] LIB gn_lib.lib
[172/172] LINK gn.exe
Generating code
Finished generating code
	D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.gn.Release
	cd core\ && ( if not exist Makefile D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -o Makefile D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\core\core.pro "CONFIG-=debug" "CONFIG+=release" "CONFIG-=static" "CONFIG*=shared" -qtconf D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/qt.conf ) && D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile
	( if not exist Makefile.core_headers D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -o Makefile.core_headers D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\core\core_headers.pro "CONFIG-=debug" "CONFIG+=release" "CONFIG-=static" "CONFIG*=shared" -qtconf D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/qt.conf ) && D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.core_headers
	D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.core_headers.Release
	( if not exist Makefile.core_generator D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -o Makefile.core_generator D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\core\core_generator.pro "CONFIG-=debug" "CONFIG+=release" "CONFIG-=static" "CONFIG*=shared" -qtconf D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/qt.conf ) && D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.core_generator
	D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.core_generator.Release
	( if not exist Makefile.gn_run D:\app\vcpkg\installed\x64-windows\tools\qt5\bin\qmake.exe -o Makefile.gn_run D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\core\gn_run.pro "CONFIG-=debug" "CONFIG+=release" "CONFIG-=static" "CONFIG*=shared" -qtconf D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/qt.conf ) && D:\app\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.gn_run
ERROR at //build/toolchain/win/BUILD.gn:432:24: Script returned non-zero exit code.
  x64_toolchain_data = exec_script("setup_toolchain.py",
                       ^----------
Current dir: D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/x64-windows-rel/src/core/release/
Command: "D:\app\vcpkg\downloads\tools\python\python-2.7.16-x64\python.exe" D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/src/5.12.7-0f8fbdda3a/src/3rdparty/chromium/build/toolchain/win/setup_toolchain.py win x64 environment.x64
Returned 1 and printed out:

vc_bin_dir = "D:\\Programs\\MSVS2019\\VC\\Tools\\MSVC\\14.25.28610\\bin\\HostX64\\x64"
include_flags_I = "/ID:\\Programs\\MSVS2019\\VC\\Tools\\MSVC\\14.25.28610\\include \"/IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt\" \"/IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\" \"/IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\" \"/IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt\" \"/IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt\""
include_flags_imsvc = "-imsvcD:\\Programs\\MSVS2019\\VC\\Tools\\MSVC\\14.25.28610\\include \"-imsvcC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt\" \"-imsvcC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\" \"-imsvcC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\" \"-imsvcC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt\" \"-imsvcC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt\""
vc_lib_path = "D:\\Programs\\MSVS2019\\VC\\Tools\\MSVC\\14.25.28610\\lib\\x64"

See //BUILD.gn:62:1: which caused the file to be included.
group("gn_all") {
^----------------
```

package-build-x64-windows-rel-err.log :

```console
jom 1.1.3 - empower your cores

Project MESSAGE: Running: D:\app\vcpkg\buildtrees\qt5-webengine\x64-windows-rel\src\3rdparty\gn\out\Release\gn.exe gen D:\app\vcpkg\buildtrees\qt5-webengine\x64-windows-rel\src\core\release --script-executable=D:\app\vcpkg\downloads\tools\python\python-2.7.16-x64\python.exe --args="use_qt=true is_component_build=false is_shared=true enable_message_center=false enable_mus=false enable_nacl=false enable_remoting=false enable_reporting=false enable_web_speech=false enable_widevine=true has_native_accessibility=false use_allocator_shim=false use_allocator=\"none\" v8_use_external_startup_data=false treat_warnings_as_errors=false enable_swiftshader=false use_custom_libcxx=false enable_basic_printing=true enable_print_preview=true enable_pdf=true enable_plugins=true enable_spellcheck=true enable_webrtc=true enable_precompiled_headers=false is_official_build=true is_debug=false symbol_level=0 remove_webcore_debug_symbols=true remove_v8base_debug_symbols=true v8_use_snapshot=true use_kerberos=true is_clang=false use_sysroot=false enable_session_service=false ninja_use_custom_environment_files=false is_multi_dll_chrome=false win_linker_timing=true com_init_check_hook_disabled=true is_win_fastlink=false use_incremental_linking=false visual_studio_version=2019 visual_studio_path=\"D:/Programs/MSVS2019\" windows_sdk_path=\"C:/Program Files ^(x86^)/Windows Kits/10\" target_cpu=\"x64\" qtwebengine_target=\"D:\app\vcpkg\buildtrees\qt5-webengine\x64-windows-rel\src\core\release:QtWebEngineCore\"" --root=D:\app\vcpkg\buildtrees\qt5-webengine\src\5.12.7-0f8fbdda3a\src\3rdparty\chromium 
Traceback (most recent call last):
  File "D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/src/5.12.7-0f8fbdda3a/src/3rdparty/chromium/build/toolchain/win/setup_toolchain.py", line 303, in <module>
    main()
  File "D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/src/5.12.7-0f8fbdda3a/src/3rdparty/chromium/build/toolchain/win/setup_toolchain.py", line 290, in main
    assert vc_lib_atlmfc_path
AssertionError
Traceback (most recent call last):
  File "D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/src/5.12.7-0f8fbdda3a/src/3rdparty/chromium/build/toolchain/win/setup_toolchain.py", line 303, in <module>
    main()
  File "D:/bluehawk/bhawk/vcpkg/buildtrees/qt5-webengine/src/5.12.7-0f8fbdda3a/src/3rdparty/chromium/build/toolchain/win/setup_toolchain.py", line 290, in main
    assert vc_lib_atlmfc_path
AssertionError
Project ERROR: GN run error!
jom: D:\app\vcpkg\buildtrees\qt5-webengine\x64-windows-rel\src\core\Makefile [sub-gn_run-pro-make_first] Error 3
jom: D:\app\vcpkg\buildtrees\qt5-webengine\x64-windows-rel\src\Makefile [sub-core-make_first] Error 2
jom: D:\app\vcpkg\buildtrees\qt5-webengine\x64-windows-rel\Makefile [sub-src-make_first] Error 2
```

**Additional context**
Building qt5-webengine:x64-windows fails with `Project ERROR: GN run error!` this is for Qt 5.12.7 (The default version of Qt in vcpkg.)

The same error also occurs if `vcpkg install qt5-base[latest]:x64-windows qt5-webengine:x64-windows` is used instead. But there's more. When `qt5-base[latest]:x64-windows` is selected, the build will fail before the error reported here and will do so at the point when the patches are applied to the extracted source archive. The folks at Qt decided to change the locations of some of the files which need to be patched in Qt version 5.14.2. I was able to work around this to check if the issue reported here persisted to the latest Qt version and it does.

My guess here is that gn is looking for vc_lib_atlmfc_path. According to https://doc.qt.io/qt-5/qtwebengine-platform-notes.html#windows 

> On Windows, the following additional tools are required:
Active Template Library (ATL), usually included in the Visual Studio installation

I tried building with and without ATL installed and the build error still persists. Although I didn't add it to the environment path. If the atlmfc port needs to be added to the control file as a dependency, is there a way atl needs to be added to the build environment?

I'm fairly sure this issue is unrelated to https://github.com/microsoft/vcpkg/issues/11019 as, from my testing it happens to qt5-base[latest] also.

@Neumann-A

> Did you also select MFC in the VS Installer?

Yep:

![git](https://user-images.githubusercontent.com/186629/80770697-d6732d00-8b48-11ea-9298-d243638000a5.png)

> but I know that it wont finish since my machine does not have enough RAM for the rel build

https://chromium.googlesource.com/chromium/src/+/master/docs/windows_build_instructions.md

> A 64-bit Intel machine with at least 8GB of RAM.

4GB Surface Pro? https://www.microsoft.com/en-us/p/surface-pro/8nkt9wttrbjk?activetab=pivot:techspecstab

And:
https://github.com/Neumann-A/vcpkg/blob/7d7e0591e702e6e52cb0d8dbabcbf1dbfb0c100a/ports/qt5-webengine/portfile.cmake#L22

`get_filename_component(NINJA_DIR "${GPERF}" DIRECTORY )`
Do you mean? :
`get_filename_component(NINJA_DIR "${NINJA}" DIRECTORY )`

Also, I've been using powershell instead of patch files here:

```cmake
## Timestamp announce to console
string(TIMESTAMP B_TS1 "Build started at %H:%M:%S on %d-%m-%y")
message(STATUS "* Build QtWebEngine from source begin:")
message(STATUS "* ${B_TS1}")
message(STATUS "* The process will take several hours")
message(STATUS "* (4-6 on average) to complete.")
message(STATUS "* Please be patient whilst the software is built")

if(QT_MAJOR_MINOR_VER STREQUAL "5.14")

    qt_download_submodule(OUT_SOURCE_PATH TARGET_SOURCE_PATH ${ARGV})

    if(EXISTS ${TARGET_SOURCE_PATH}/src/3rdparty/chromium/README.md)

        message(STATUS "* Patching the source tree...")
        ### ------------------------------------------------------
        ##                  "Fixes"
        ### ------------------------------------------------------

        file(READ ${TARGET_SOURCE_PATH}/src/buildtools/config/common.pri COM_PRI)
        ## "Fix" the "Ninja does not see existing precompiled header" error https://bugreports.qt.io/browse/QTBUG-65677
        ## Compiling the headers means the build takes significantly longer to complete
        string(REPLACE "gn_args += enable_precompiled_headers=true" "gn_args += enable_precompiled_headers=false" COM_PRI "${COM_PRI}")
        ##
        ## Write out the file
        file(WRITE ${TARGET_SOURCE_PATH}/src/buildtools/config/common.pri "${COM_PRI}")
        ## "Fix" the "chromium\ui\gl\gl_bindings_autogen_gl.h(401): error C2061: syntax error: identifier 'GLeglImageOES'" error message
        ## when compiling without precompiled headers
        file(READ ${TARGET_SOURCE_PATH}/src/3rdparty/chromium/ui/gl/gl_bindings_autogen_gl.h GLB_H)
	    string(REPLACE "class GLContext;" "class GLContext;\n\ntypedef void *GLeglImageOES;" GLB_H "${GLB_H}")
	    file(WRITE ${TARGET_SOURCE_PATH}/src/3rdparty/chromium/ui/gl/gl_bindings.h "${GLB_H}")
        file(REMOVE_RECURSE ${TARGET_SOURCE_PATH}/src/3rdparty/ninja) # QtWebEngine ships with a version of ninja, comment this line out if you want to use it

        ### ------------------------------------------------------

        ## Prevent patching twice upon rebuilds
        file(REMOVE ${TARGET_SOURCE_PATH}/src/3rdparty/chromium/README.md)

        message(STATUS "* Source tree patched")

    else()

        message(STATUS "* Source tree is already patched...")

    endif()
    ## Add some includes to the INCLUDEPATH
    set(ENV{INCLUDEPATH} "${TARGET_SOURCE_PATH}/include/;$ENV{INCLUDEPATH}")
    set(ENV{PATH} "${TARGET_SOURCE_PATH}/src/3rdparty/gn/);$ENV{PATH}")

    if(QT_UPDATE_VERSION)
        SET(VCPKG_POLICY_EMPTY_PACKAGE enabled PARENT_SCOPE)
    else()
        qt_build_submodule(${TARGET_SOURCE_PATH} ${ARGV})
        qt_install_copyright(${TARGET_SOURCE_PATH})
    endif()

else()

qt_submodule_installation(PATCHES 
                                common.pri.patch
                                gl.patch)

endif()

string(TIMESTAMP B_TS2 "Build completed at %H:%M:%S on %d-%m-%y")
message(STATUS "* ${B_TS2}")
```

--- a/ports/qt5-webengine/CONTROL
+++ b/ports/qt5-webengine/CONTROL
@@ -1,5 +1,5 @@
 Source: qt5-webengine
 Version: 5.12.8
 Description: Qt5 webengine Module; 
-Build-Depends:  qt5-base, qt5-declarative, qt5-location, qt5-quickcontrols, qt5-quickcontrols2, qt5-tools, qt5-webchannel, atlmfc (windows)
+Build-Depends:  qt5-base, qt5-declarative, qt5-location, qt5-quickcontrols, qt5-quickcontrols2, qt5-tools, qt5-webchannel, atlmfc (windows), ffmpeg[core] (!windows)
 Supports: !static

--- a/ports/qt5-webengine/CONTROL
+++ b/ports/qt5-webengine/CONTROL
@@ -1,5 +1,5 @@
 Source: qt5-webengine
-Version: 5.12.7-1
+Version: 5.12.8
 Description: Qt5 webengine Module; 
-Build-Depends:  qt5-base, qt5-declarative, qt5-location, qt5-quickcontrols, qt5-quickcontrols2, qt5-tools, qt5-webchannel
+Build-Depends:  qt5-base, qt5-declarative, qt5-location, qt5-quickcontrols, qt5-quickcontrols2, qt5-tools, qt5-webchannel, atlmfc (windows)
 Supports: !static

--- a/ports/qt5-webengine/common.pri.latest.patch
+++ b/ports/qt5-webengine/common.pri.latest.patch
@@ -1,0 +1,12 @@
+diff --git a/src/buildtools/config/common.pri b/src/buildtools/config/common.pri
+#latest diff --git a/src/buildtools/config/common.pri b/src/buildtools/config/common.pri
+index 97d39535c..dc09da43a 100644
+--- a/src/buildtools/config/common.pri	
++++ b/src/buildtools/config/common.pri
+@@ -57,5 +57,5 @@
+ precompile_header {
+-    gn_args += enable_precompiled_headers=true
++    gn_args += enable_precompiled_headers=false
+ } else {
+     gn_args += enable_precompiled_headers=false
+ }

--- a/ports/qt5-webengine/gl.latest.patch
+++ b/ports/qt5-webengine/gl.latest.patch
@@ -1,0 +1,13 @@
+diff --git a/src/3rdparty/chromium/ui/gl/gl_bindings_autogen_gl.h b/src/3rdparty/chromium/ui/gl/gl_bindings_autogen_gl.h
+index d54583e8b..1921185f8 100644
+--- a/src/3rdparty/chromium/ui/gl/gl_bindings_autogen_gl.h	
++++ b/src/3rdparty/chromium/ui/gl/gl_bindings_autogen_gl.h
+@@ -16,7 +16,7 @@
+ namespace gl {
+ 
+ class GLContext;
+-
++typedef void *GLeglImageOES;
+ typedef void(GL_BINDING_CALL* glActiveShaderProgramProc)(GLuint pipeline,
+                                                          GLuint program);
+ typedef void(GL_BINDING_CALL* glActiveTextureProc)(GLenum texture);

--- a/ports/qt5-webengine/portfile.cmake
+++ b/ports/qt5-webengine/portfile.cmake
@@ -10,6 +10,8 @@ if(NOT VCPKG_TARGET_IS_WINDOWS)
 endif()
 include(${CURRENT_INSTALLED_DIR}/share/qt5/qt_port_functions.cmake)
 
+
+
 vcpkg_find_acquire_program(FLEX)
 vcpkg_find_acquire_program(BISON)
 vcpkg_find_acquire_program(GPERF)
@@ -19,7 +21,7 @@ get_filename_component(FLEX_DIR "${FLEX}" DIRECTORY )
 get_filename_component(BISON_DIR "${BISON}" DIRECTORY )
 get_filename_component(PYTHON2_DIR "${PYTHON2}" DIRECTORY )
 get_filename_component(GPERF_DIR "${GPERF}" DIRECTORY )
-get_filename_component(NINJA_DIR "${GPERF}" DIRECTORY )
+get_filename_component(NINJA_DIR "${NINJA}" DIRECTORY )
 
 if(WIN32) # WIN32 HOST probably has win_flex and win_bison!
     if(NOT EXISTS "${FLEX_DIR}/flex${VCPKG_HOST_EXECUTABLE_SUFFIX}")
@@ -37,9 +39,13 @@ vcpkg_add_to_path(PREPEND "${GPERF_DIR}")
 vcpkg_add_to_path(PREPEND "${NINJA_DIR}")
 
 if(QT_MAJOR_MINOR_VER STREQUAL "5.14")
-    set(PATCHES common.pri.latest.patch)
+    set(PATCHES common.pri.latest.patch gl.latest.patch)
 else()
     set(PATCHES common.pri.patch gl.patch)
 endif()
 
-qt_submodule_installation(PATCHES ${PATCHES} )
+if(NOT VCPKG_TARGET_IS_WINDOWS)
+    list(APPEND CORE_OPTIONS "BUILD_OPTIONS" "-webengine-system-libwebp" "-webengine-system-ffmpeg" "-webengine-system-icu")
+endif()
+
+qt_submodule_installation(${CORE_OPTIONS} PATCHES ${PATCHES})

--- a/ports/qt5-webengine/portfile.cmake
+++ b/ports/qt5-webengine/portfile.cmake
@@ -36,6 +36,10 @@ vcpkg_add_to_path(PREPEND "${PYTHON2_DIR}")
 vcpkg_add_to_path(PREPEND "${GPERF_DIR}")
 vcpkg_add_to_path(PREPEND "${NINJA_DIR}")
 
-qt_submodule_installation(PATCHES 
-                                common.pri.patch
-                                gl.patch)
+if(QT_MAJOR_MINOR_VER STREQUAL "5.14")
+    set(PATCHES common.pri.latest.patch)
+else()
+    set(PATCHES common.pri.patch gl.patch)
+endif()
+
+qt_submodule_installation(PATCHES ${PATCHES} )

